src/driver/Driver.ts

```diff
@@ -115,6 +120,11 @@ export interface Driver {
     normalizeIsUnique(column: ColumnMetadata): boolean;
 
     /**
+     * Calculates column length taking into account the default length values.
+     */
+    getColumnLength(column: ColumnMetadata): number | string | undefined;
```

Comment:
Do you really think we shall deal with `number|string` pain everywhere? Maybe we just cast into into string (for example when we read from decorator or somewhere in column metadata, I don't know) and use that string everywhere then?

Reply:
I am a bit worried about such a change. 
Now, the only reason length is string is to allow the "ΜΑΧ" value (or similar) and this could have been avoided using a convention like -1 for max, still the type would have been integer and we could handle this inside each driver (although it might look a bit strange or even create problems in some cases).

What worries me is that at some places we pass length to the actual db driver(s) and they might expect this to be number not string. Now we convert it to number where it is possible so the property type would be correct. In the other case we might have to check a ton of scenarios. 

I have not looked into each driver in depth, this would take huge amount of time to get into such details.

It is still your call, what would you like me to do with this ?

Reply:
> What worries me is that at some places we pass length to the actual db driver(s) and they might expect this to be number not string.

we do not pass length anywhere into underlying db drivers. We use length only internally to concatenate length to column type. Also we use length to compare things to understand if column has changed.

Reply:
How about MsSqlParameter ? This also passes the length implicitly i believe ...

Reply:
not sure can you check that, anyway Im sure its better to pass things in a single format then in two formats

Reply:
Ok I will change it

Reply:
yeah lets see what will be